pkg/kubelet/nodelease/controller.go

```diff
@@ -140,18 +145,23 @@ func (c *controller) ensureLease() (*coordv1beta1.Lease, bool, error) {
 
 // retryUpdateLease attempts to update the lease for maxUpdateRetries,
 // call this once you're sure the lease has been created
-func (c *controller) retryUpdateLease(base *coordv1beta1.Lease) {
+func (c *controller) retryUpdateLease(base *coordv1beta1.Lease) error {
+	var err error
 	for i := 0; i < maxUpdateRetries; i++ {
-		_, err := c.leaseClient.Update(c.newLease(base))
+		// OptimisticLockError requires getting the newer version of lease to proceed.
```

Comment:
Move it to the bottm of the loop - then you don't need to define err, and obviously in the first iteration it will always be nil.

Reply:
Done.
I wanted to avoid additional get call. I assume that code is more readable now?